syft/pkg/cataloger/apkdb/parse_apk_db.go

```diff
@@ -21,7 +23,7 @@ var _ common.ParserFn = parseApkDB
 
 // parseApkDb parses individual packages from a given Alpine DB file. For more information on specific fields
 // see https://wiki.alpinelinux.org/wiki/Apk_spec .
-func parseApkDB(_ string, reader io.Reader) ([]pkg.Package, error) {
+func parseApkDB(_ string, reader io.Reader) ([]pkg.Package, []artifact.Relationship, error) {
```

Comment:
Are we adding relationships as a placeholder for now? On parse what relationships are we expecting to be able to generate?

Reply:
Yeah, a necessary placeholder for now 👍 

Since the generic cataloger has been enhanced to [allow for parser functions to raise up artifact relationships](https://github.com/anchore/syft/blob/aba0de6e651d17a6e291e144cb92bdfd04693c69/syft/pkg/cataloger/common/parser.go#L11), and the APK cataloger [uses the generic cataloger](https://github.com/anchore/syft/blob/aba0de6e651d17a6e291e144cb92bdfd04693c69/syft/pkg/cataloger/apkdb/cataloger.go#L17), that meant that this function signature needed to be updated as well.

Reply:
There may not be any relationships that the APK DB parser will raise up, but other parser functions may have the opportunity to raise up relationships (such as `gemfile.lock` and `rpmdb` parsers). I haven't double checked the underlying data available for APK yet, but APK might have package dependency data for us to leverage.